app-creator/client/src/widgets/story-board/story-board.ts

```diff
@@ -102,49 +112,206 @@ export class Storyboard extends connect(store)(LitElement) {
       store.dispatch(setSelectedTemplateID(template.config.id));
 
-      const { storyboard } = this;
-      if (storyboard) {
-        storyboard.innerHTML = ``;
-        generateUI(template, storyboard);
-      }
+      this.clearSelectedPalette();
 
-      this.requestUpdate();
+      this.renderNewTemplate(template);
+    }
+
+    /**
+     * We want to Re-render the storyboard when we switch the template color palette.
+     * We do this by checking if the changingPalette event has been emitted.
+     */
+    if (state.eventType === EventType.changingPalette) {
+      store.dispatch(setEventType(EventType.none, true));
+
+      this.renderNewTemplate(template);
     }
   }
 
   /**
-   * Additional custom styles.
+   * Clears the value of the palette select input and restores initial state.
    */
-  @property({ type: Object }) styles = {};
+  clearSelectedPalette() {
+    if (this.paletteSelect) {
+      this.paletteSelect.value = '';
+    }
+    this.selectedPalette = Palette.light;
+    this.requestedPalette = Palette.light;
+  }
 
   /**
-   * Reference to storyboard element.
+   * Renders new template by calling generateUI and requesting update.
    */
-  @query(`#${STORYBOARD_ID}`) storyboard!: PaperCardElement;
+  renderNewTemplate(template: AppCreatorStore['template']) {
+    const { storyboard } = this;
+    if (storyboard) {
+      storyboard.innerHTML = ``;
+      generateUI(template, storyboard);
+    }
+
+    this.requestUpdate();
+  }
+
+  /**
+   * Sets the selected palette in state and opens the confirmation dialog.
+   */
+  handlePaletteChange(e: Event) {
+    this.requestedPalette = (e.target as HTMLSelectElement).value as Palette;
+    if (this.paletteConfimationDialog) {
+      this.paletteConfimationDialog.open();
+    }
+  }
+
+  /**
+   * Called when users confirm on the confirmation dialog when changing palettes.
+   */
+  paletteChangeConfirmation() {
+    this.selectedPalette = this.requestedPalette;
+    store.dispatch(setPalette(this.selectedPalette));
+    store.dispatch(setEventType(EventType.changingPalette, true));
+  }
+
+  /**
+   * Called when users decline palette change.
+   */
+  paletteChangeDecline() {
+    if (this.paletteConfimationDialog) {
+      this.paletteConfimationDialog.close();
+    }
+    if (this.paletteSelect) {
+      this.paletteSelect.value = this.selectedPalette;
+      this.requestedPalette = this.selectedPalette;
+    }
+  }
 
   render() {
-    const { styles } = this;
+    const {
+      styles,
+      handlePaletteChange,
+      paletteChangeConfirmation,
+      paletteChangeDecline,
+    } = this;
 
     const isMobile =
       store.getState().template.config?.device === DeviceType.mobile;
 
+    const paletteSelectInput = html`
+      <div id="palette-select-container">
+        <select
+          name="Select palette"
+          id="select-palette"
+          @change=${handlePaletteChange}
+        >
+          <option value="" disabled selected>Change Palette</option>
```

Comment:
It's better to have a proper label next to the select instead of using a disabled option. It also means you can set the default option to the one currently selected on the template!